packages/nx-plugin/src/generators/create-package/create-package.ts

```diff
@@ -0,0 +1,214 @@
+import {
+  addDependenciesToPackageJson,
+  readProjectConfiguration,
+  Tree,
+  generateFiles,
+  readJson,
+  convertNxGenerator,
+  formatFiles,
+  updateProjectConfiguration,
+  updateJson,
+  GeneratorCallback,
+  runTasksInSerial,
+  workspaceRoot,
+  joinPathFragments,
+} from '@nx/devkit';
+import { libraryGenerator as jsLibraryGenerator } from '@nx/js';
+import { nxVersion } from '../../utils/versions';
+import generatorGenerator from '../generator/generator';
+import { CreatePackageSchema } from './schema';
+import { NormalizedSchema, normalizeSchema } from './utils/normalize-schema';
+import e2eProjectGenerator from '../e2e-project/e2e';
+import { getPluginCapabilities } from 'nx/src/utils/plugins/plugin-capabilities';
+
+export async function createPackageGenerator(
+  host: Tree,
+  schema: CreatePackageSchema
+) {
+  const tasks: GeneratorCallback[] = [];
+
+  const options = normalizeSchema(host, schema);
+  const pluginPackageName = await addPresetGenerator(host, options);
+
+  const installTask = addDependenciesToPackageJson(
+    host,
+    {
+      'create-nx-workspace': nxVersion,
+    },
+    {}
+  );
+  tasks.push(installTask);
+
+  await createCliPackage(host, options, pluginPackageName);
+  if (options.e2eTestRunner !== 'none') {
+    tasks.push(await addE2eProject(host, options));
+  }
+
+  if (!options.skipFormat) {
+    await formatFiles(host);
+  }
+
+  return runTasksInSerial(...tasks);
+}
+
+/**
+ * Add a preset generator to the plugin if it doesn't exist
+ * @param host
+ * @param schema
+ * @returns package name of the plugin
+ */
+async function addPresetGenerator(
+  host: Tree,
+  schema: NormalizedSchema
+): Promise<string> {
+  const { root: projectRoot } = readProjectConfiguration(host, schema.project);
+  const plugin = await getPluginCapabilities(workspaceRoot, schema.project);
+  if (!plugin?.generators['preset']) {
+    await generatorGenerator(host, {
```

Comment:
@AgentEnder Make this generator error

@xiongemi Catch the error. Check the type of error.